### PR TITLE
docs: Initial Next.js Coding Standards Documentation

### PR DESCRIPTION
**Pull request by mlalivio**
## What does this PR do?
- Introduces initial coding standards for Next.js projects

## Why did you choose this solution?
- To provide a guideline for project structure and coding practices

## What is the dependency graph of the changes?
- Not applicable

## How to test/verify?
- Review the documentation standards added

## Related ticket url:
-

### CODE REVIEW

**Comment by symphdevtools** [comment]
Hi! I'm generating PR title and description for you and will automatically apply it to your PR. I'll react :rocket: to this comment when done. Just sit back and chill!

When you have new changes to this PR and you want to regenerate the title and PR description, just comment `churva`!

Let me know if something is not working well, e.g.:
- Some parts of the title and description weren't preserved when generating title and description
- Disappearing screenshots or links
- Etc.

Thanks!

(RIP hatdog)

File: standards/nextjs_standards.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## File Structure & Naming Convention
+
+- Organize project files under the `src` directory for clarity.
+- Use the `pages` directory for page components, leveraging Next.js's file-based routing.
+- Store static assets in the `public` directory.
+- Group components into a `components` directory, categorizing further if necessary.
+
+## Component Structure
+
+Use atomic design when organizing components. Organize components into atoms, molecules, organisms, and templates for scalability and maintainability.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:45]: Let's put speicific definitions and samples what atoms, molecules, organisms, and templates are

File: standards/nextjs_standards.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+- Leverage Next.js's built-in code splitting to manage application code in smaller, more manageable units.
+- Utilize dynamic imports for loading non-essential components on demand, reducing the initial bundle size and improving page load times.
+- Identify components or code segments not critical for the initial page load and load them asynchronously.
+
+## Rendering Modes
+
+- **Server-Side Rendering (SSR)**: Use for pages requiring dynamic content or SEO, ideal for public-facing and content-heavy pages.
+- **Static Site Generation (SSG)**: Use for content with infrequent changes, suitable for landing pages, blogs, and marketing sites.
+- **Client-Side Rendering (CSR)**: Use for interactive components and real-time data, prioritizing pages/components that require interactivity.
+- **Hybrid Rendering**: Combine SSR and CSR benefits for applications with mixed content and interactivity needs.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:58]: Let's add examples

File: standards/nextjs_standards.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+## Styling
+
+- Utilize CSS Modules for component-level styles to avoid conflicts.
+- Consider CSS-in-JS solutions like styled-components for dynamic styling.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:63]: We always prefer tailwindcss, let's remove these 2

File: standards/nextjs_standards.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+- Optimize images and leverage Next.js's image optimization.
+- Use static generation (SSG) where possible for faster page loads.
+- Analyze and optimize bundle sizes with Next.js's built-in analytics.
+
+## Security Practices
+
+- Sanitize user input to prevent XSS attacks.
+- Use environment variables for sensitive information.
+- Regularly update dependencies to mitigate vulnerabilities.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:83]: Add samples like for sanitize, let's add our preferred our way and library to do this

File: standards/nextjs_standards.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+## Testing
+
+- Write unit and integration tests using tools like Jest and React Testing Library.
+- Aim for a high coverage to ensure reliability and maintainability.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:88]: Add our minimum requirement for test coverage

File: standards/nextjs_standards.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+## Component Structure
+
+Use atomic design when organizing components. Organize components into atoms, molecules, organisms, and templates for scalability and maintainability.
+
+## Efficient Code Splitting, Dynamic Imports, and Loading of Client-Side Scripts
+
+- Leverage Next.js's built-in code splitting to manage application code in smaller, more manageable units.
+- Utilize dynamic imports for loading non-essential components on demand, reducing the initial bundle size and improving page load times.
+- Identify components or code segments not critical for the initial page load and load them asynchronously.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:51]: Add examples

File: standards/nextjs_standards.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+- Use an active voice for readability.
+- Avoid subjective words like "easy", "quick", "simple", "just", etc.
+- Write in the second person (you/your) and use gender-neutral language.
+- Ensure code examples are properly formatted and working2.
+
+## Code Formatting
+
+- Use Prettier for automatic code formatting to maintain consistency.
+- Configure ESLint with Next.js specific rules to catch common issues and enforce style guidelines.
+- Adopt a consistent indentation method (e.g., 2 spaces) across the project.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:27]: Let's also add that we prefer airbnb style guide

File: standards/nextjs_standards.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+- **Performance**: Optimize for speed and efficiency, considering the performance implications of your code.
+
+## Writing
+
+- Write clear, concise sentences and avoid tangents.
+- Use simple words; for example, use "use" instead of "utilize".
+- Use an active voice for readability.
+- Avoid subjective words like "easy", "quick", "simple", "just", etc.
+- Write in the second person (you/your) and use gender-neutral language.
+- Ensure code examples are properly formatted and working2.
```
Comment (JHNLWHD) [inline standards/nextjs_standards.md:21]: Add example codes

**Review by JHNLWHD — CHANGES_REQUESTED** [review]

File: standards/nextjs_standards.md (hunk 1 above)
Comment (igndev10) [reply standards/nextjs_standards.md:45]: Got this 👌🏼

File: standards/nextjs_standards.md (hunk 2 above)
Comment (igndev10) [reply standards/nextjs_standards.md:58]: Got this 👌🏼

File: standards/nextjs_standards.md (hunk 3 above)
Comment (igndev10) [reply standards/nextjs_standards.md:63]: Ah yes, will update this

File: standards/nextjs_standards.md (hunk 4 above)
Comment (igndev10) [reply standards/nextjs_standards.md:83]: Got this 👌🏼

File: standards/nextjs_standards.md (hunk 5 above)
Comment (igndev10) [reply standards/nextjs_standards.md:88]: okay 👌🏼

File: standards/nextjs_standards.md (hunk 6 above)
Comment (igndev10) [reply standards/nextjs_standards.md:51]: Got this 👌🏼

File: standards/nextjs_standards.md (hunk 7 above)
Comment (igndev10) [reply standards/nextjs_standards.md:27]: Got this 👌🏼

File: standards/nextjs_standards.md (hunk 8 above)
Comment (igndev10) [reply standards/nextjs_standards.md:21]: Got this 👌🏼

**Review by JHNLWHD — APPROVED** [review]
LGTM 👍🏼 

Comments will be resolved on a separate PR by @iannatividad

**Pull request merged**